config/import-control.xml

```diff
@@ -14,7 +14,6 @@
   <!-- guava causes issues with other libraries since not everyone is on the same version -->
   <disallow pkg="com\.google.*" regex="true"/>
 
-  <!-- until https://github.com/checkstyle/checkstyle/issues/6628 -->
   <allow class="com.puppycrawl.tools.checkstyle.DetailAstImpl"/>
```

Comment:
@rnveach this item is still valid after #6628 closed.
Should there be some other issue?

Reply:
@romani Please help me remember this.

I assume the reason this class is allowed, but is expected to be removed is so only very specific areas deal with `DetailAstImpl` and we want everything to use the interface, DetailAST`, by default?

Reply:
yes, git blame points to https://github.com/checkstyle/checkstyle/pull/6629
the best way to get to know current status is follow "until" and remove this `allow`.
Comment explain a code not a reason of comment itself.

Reply:
@romani Removal of this line causes violations. This is the reason why I asked about this line

Reply:
```
✘-1 ~/java/github/romani/checkstyle [master|✚ 1] 
$ git diff
diff --git a/config/import-control.xml b/config/import-control.xml
index 4b52e85..a5300b4 100644
--- a/config/import-control.xml
+++ b/config/import-control.xml
@@ -14,9 +14,6 @@
   <!-- guava causes issues with other libraries since not everyone is on the same version -->
   <disallow pkg="com\.google.*" regex="true"/>
 
-  <!-- until https://github.com/checkstyle/checkstyle/issues/6628 -->
-  <allow class="com.puppycrawl.tools.checkstyle.DetailAstImpl"/>
-
   <allow pkg="com.puppycrawl.tools.checkstyle.api"/>
   <allow pkg="com.puppycrawl.tools.checkstyle.checks"/>
   <allow pkg="java.io"/>

 $ mvn compile antrun:run@ant-phase-verify

[INFO] --- maven-antrun-plugin:3.0.0:run (ant-phase-verify) @ checkstyle ---
[INFO] Executing tasks
[INFO]      [echo] Checkstyle started (checkstyle_checks.xml): 12/02/2021 05:06:26 AM
[INFO] [checkstyle] Running Checkstyle  on 1403 files
[ERROR] [checkstyle] [ERROR] /home/rivanov/java/github/romani/checkstyle
/src/main/java/com/puppycrawl/tools/checkstyle/checks/blocks/RightCurlyCheck.java:25:1: 
Disallowed import - com.puppycrawl.tools.checkstyle.DetailAstImpl. [ImportControlMain]

....
Checkstyle failed: Got 1 errors and 0 warnings.

```

Reply:
not that old problem - https://github.com/checkstyle/checkstyle/commit/1e528d6988aa11c9b06caa7daf88d3edec0add6f#diff-926a98644a89150935e8cd4c38d9a432dbef54fad501b647c073dbce50e7d849R535
Very messy PR - https://github.com/checkstyle/checkstyle/pull/4247 where I user admin power to merge a mess.

Reply:
@strkkk , please create new issue to resolve this problem.

Reply:
@romani ok, what about https://github.com/checkstyle/checkstyle/pull/9264#discussion_r574166403?

Reply:
Created https://github.com/checkstyle/checkstyle/issues/9271
I will update reference

Reply:
Thanks.